src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs

```diff
@@ -91,7 +91,13 @@ public static PaymentVerifyResult CreateVerifyResult(string response, MessagesOp
                 return PaymentVerifyResult.Failed(message);
             }
 
-            return PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
+            var res = PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
+            res.AdditionalData.Add("FactorNumber", result.FactorNumber);
+            res.AdditionalData.Add("Mobile", result.Mobile);
+            res.AdditionalData.Add("Description", result.Description);
+            res.AdditionalData.Add("CardNumber", result.CardNumber);
```

Comment:
Gateways should have their own Additional Data class (Holder). We should not use `string` keys.
Why:
- If we change these keys in future, then nobody can get them anymore.
- Users have to know what exactly the `keys` are.
- Keys are Case-Sensitive.
- Users have to write 4 lines of codes for getting 4 values.

So create a separate class something like: `PayIrAdditionalData` inside the main folder of this gateway.
Put these properties in it.
Create an extension method for getting the additional data from the IPaymentVerifyResult interface.
```c#
namespace Parbad.Gateway.PayIr
{
   public static class PayIrAdditionalDataExtensions
   {
      private const string AdditionalDataKey = "PayIrAdditionalData";

      public static PayIrAdditionalData GetPayIrAdditionalData(this IPaymentVerifyResult result)
      {
         return (PayIrAdditionalData) result.AdditionalData[AdditionalDataKey];
      }
   }
}
```